src/macros/grab_value.cc

```diff
@@ -242,7 +242,7 @@ static int handle_service_macro(
       service_map::const_iterator
         found(service::services.find({arg1, arg2}));
 
-      if (found == service::services.end() || !found->second)
+      if (found != service::services.end() && found->second)
```

Comment:
This bug has already been fixed previously with hosts.
Could you add a unit test to check the good behavior, please?